src/framework/queries/patrol.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  GameState,
+  Instance,
+} from "../types";
+
+import {
+  getPlayer,
+} from "./common";
+
+export const isScavenger = (
+  $: GameState,
+  I: Instance,
+): boolean => {
+  const P = getPlayer($, I.controller);
+  if (P == null || $.activePlayer === P.id) {
```

Comment:
I think id would be a helpful field to have on PlayerState, but you could use I.controller instead.

Reply:
Correct, I left this in (knowing it currently breaks) because iirc I already have a PR somewhere that adds it. There are a lot of usecases where being able to backtrack to get a player pointer's ID is important.